lib/libzutil/zutil_import.c

```diff
@@ -929,22 +1006,39 @@ zpool_read_label(int fd, nvlist_t **config, int *num_labels)
 
 	if (lio_listio(LIO_WAIT, aiocbps, VDEV_LABELS, NULL) != 0) {
 		int saved_errno = errno;
-
-		if (errno == EAGAIN || errno == EINTR || errno == EIO) {
-			/*
-			 * A portion of the requests may have been submitted.
-			 * Clean them up.
-			 */
-			for (l = 0; l < VDEV_LABELS; l++) {
-				errno = 0;
-				int r = aio_error(&aiocbs[l]);
-				if (r != EINVAL)
+		errno = 0;
+		int do_slow = 0;
+		error = -1;
+
+		for (l = 0; l < VDEV_LABELS; l++) {
+			switch (aio_error(&aiocbs[l])) {
+				case 0:
+					break;
+				case -1:
+					break;
+				case EOPNOTSUPP:
+					do_slow = 1;
+				case EIO:
+				case EINTR:
+				case EAGAIN:
+					/*
+					 * A portion of the requests may have
```

Comment:
I would move this comment up.  It applies whenever `lio_listio` returns nonzero.

Reply:
Makes sense. Done.